socorro/app/fetch_transform_save_app.py

```diff
@@ -104,32 +110,140 @@ def get_application_defaults():
+
+    #--------------------------------------------------------------------------
+    def _create_iter(self):
+        # the actual mechanism of creating the iterator to be overridden in
+        # subclasses based on what they want to iterate over.  In this default,
+        # iteration will come from the crashstorage instance defined as
+        # "source".  This works for the crashmover where the generally used
+        # source is the file system walking class.
+        return self.source.new_crashes()
+
+    #--------------------------------------------------------------------------
+    def _action_between_each_iteration(self):
+        """an action to take after an item has been yielded by the iterator.
+        This method is to be overridden by derived classes"""
+        pass
 
     #--------------------------------------------------------------------------
-    def source_iterator(self):
-        """this iterator yields individual crash_ids from the source
-        crashstorage class's 'new_crashes' method."""
-        while(True):  # loop forever and never raise StopIteration
-            for x in self.source.new_crashes():
-                if x is None:
-                    yield None
-                elif isinstance(x, tuple):
-                    yield x  # already in (args, kwargs) form
-                else:
-                    yield ((x,), {})  # (args, kwargs)
+    def _action_after_iteration_completes(self):
+        """an action to be done when the iterator returned by the
+        "_create_iter" method is exhausted.  This method is to be overridden
+        in base classes that might need to do something special at that point.
+        examples may be simple logging, waiting, or resetting the iteration
+        system for the next loop."""
+        pass
+
+    #--------------------------------------------------------------------------
+    def _basic_iterator(self):
+        """this iterator yields individual crash_ids and/or Nones from the
+        iterator specified by the "_create_iter" method. Bare values yielded
+        by the "_create_iter" method get wrapped into an *args, **kwargs form.
+        That form is then used by the task manager as the arguments to the
+        worker function."""
+        for x in self._create_iter():
+            if x is None:
+                self.config.logger.debug('yielding %s', x)
+                yield None
+            elif isinstance(x, tuple):
+                self.config.logger.debug('yielding %s', x)
+                yield x  # already in (args, kwargs) form
             else:
-                yield None  # if the inner iterator yielded nothing at all,
-                            # yield None to give the caller the chance to sleep
+                arg_kwarg_form = ((x,), {})  # (args, kwargs)
+                self.config.logger.debug('yielding %s', arg_kwarg_form)
+                yield arg_kwarg_form
```

Comment:
There's a lot of repetition in this block. All for the sake of items that are not None and not a tuple. 
You could reduce it down to 
```python
for x in self._create_iter():
    if is is not None and not isinstance(x, tuple):
        x = ((x,), {})
    self.config.logger.debug('yielding %r', x)
    yield x
```
(Note also how I suggest you use `%r` instead of `%s` because it's more useful from a debugging purposes. Otherwise, `None` and `'None'` both become "None`)
```